src/components/generator/generatorFunction.js

```diff
@@ -5,57 +5,71 @@ import arePositionsSimilar from '../../logic/combos/similarPositions';
  * @param {*} conditions Conditions that the combo must meet
  * @param {number} retries Number of times to retry when generating a combo fails
  */
-export function generateRandomCombo(conditions, retries) {
-  const [
-    tricks, positions, numberOfTricks,
-    startFromCheckbox, startFromPosition,
-    allowDuplicates, allowConsecutiveTricks, allowSimilarPositions, allowTransitions, finishToFeet,
-    avgDifficulty, maxDifficulty
-  ] = conditions
-
-  if (!retries) retries = 0
+export function generateRandomCombo(conditions, retries= 0) {
+  conditions = {
+    tricks: undefined,
+    positions: [],
+    numberOfTricks: 10,
+    startFromCheckbox: false,
+    startingPositionIndex: 0,
+    allowDuplicates: false,
+    allowConsecutiveTricks: false,
+    allowSimilarPositions: true,
+    allowTransitions: false,
+    mustFinishToFeet: true,
+    avgDifficulty: -1,
+    minDifficulty: 0,
+    maxDifficulty: 11,
+    ...conditions
+  }
+
   if (retries >= 100) {
     alert("could not find a combo, try changing the settings")
     return
   }
 
-  if (numberOfTricks <= 1) {
+  if (conditions.numberOfTricks <= 1) {
     alert("You need more than one trick for a combo!")
     return
   }
 
-  let randomTricks = new Array(numberOfTricks)
-  let availableTricks = [...tricks]
+  let randomTricks = []
```

Comment:
nit: this can be a `const`.
As long as you do not overwrite the variable, e.g. 
```js
const randomTricks = [];
randomTricks = ...
```
const can be used, even when pushing / popping from the Array.

This is similar to how you can mutate the properties of a const object, but not the object itself:
```js
const x = {
   a: 0
}
// allowed
x.a = 42
// not allowed
x = {
  a: 42
}
```

Reply:
Approved